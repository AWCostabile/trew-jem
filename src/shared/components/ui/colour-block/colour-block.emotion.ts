import styled from 'react-emotion';

interface IStyledSpan {
  colour?: string;
}

export const ColouredSpan = styled('span')<IStyledSpan>`
  color: ${({ colour = 'inherit' }) => colour};
`;
