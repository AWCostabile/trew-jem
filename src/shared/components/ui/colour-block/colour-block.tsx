import * as React from 'react';
import { ColouredSpan } from './colour-block.emotion';

export type ColourText = [string] | [string, string];

interface IColourTextProps {
  text: ColourText[];
}

export const ColourBlock: React.SFC<IColourTextProps> = ({
  text: textArray
}) => (
  <React.Fragment>
    {textArray.map(([text, colour], key) => {
      const props = {
        children: key ? ` ${text}` : text,
        colour,
        key
      };
      return colour ? <ColouredSpan {...props} /> : <span {...props} />;
    })}
  </React.Fragment>
);
