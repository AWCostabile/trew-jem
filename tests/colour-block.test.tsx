import * as React from 'react';
import { create } from 'react-test-renderer';
import { ColourBlock, ColourText } from 'shared/components/ui/colour-block';

const testA: ColourText[] = [['Normal span...']];
const testB: ColourText[] = [['Coloured span!', '#C22']];
const testC: ColourText[] = [...testA, ...testB];

describe('<ColourBlock />', () => {
  it('Should render a regular span', () => {
    const tree = create(
      <div>
        <ColourBlock text={testA} />
      </div>
    );

    expect(tree).toMatchSnapshot();
  });

  it('Should render a stylised span', () => {
    const tree = create(
      <div>
        <ColourBlock text={testB} />
      </div>
    );

    expect(tree).toMatchSnapshot();
  });

  it('Should render a set of spans', () => {
    const tree = create(
      <div>
        <ColourBlock text={testC} />
      </div>
    );

    expect(tree).toMatchSnapshot();
  });
});
