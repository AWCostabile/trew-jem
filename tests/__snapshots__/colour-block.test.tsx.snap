// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColourBlock /> Should render a regular span 1`] = `
<div>
  <span>
    Normal span...
  </span>
</div>
`;

exports[`<ColourBlock /> Should render a set of spans 1`] = `
.emotion-0 {
  color: #C22;
}

<div>
  <span>
    Normal span...
  </span>
  <span
    className="emotion-0"
  >
     Coloured span!
  </span>
</div>
`;

exports[`<ColourBlock /> Should render a stylised span 1`] = `
.emotion-0 {
  color: #C22;
}

<div>
  <span
    className="emotion-0"
  >
    Coloured span!
  </span>
</div>
`;
